src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -58,457 +54,42 @@
-            break;
-        }
-        LogErrorOnFailure(err);
-    }
-
-    CHIP_ERROR ConfigureState(chip::FabricIndex fabricIndex, chip::NodeId nodeId)
-    {
-        assertChipStackLockedByCurrentThread();
-
-        if (mInitialized) {
-            // Prevent a new node connection since another is active.
-            VerifyOrReturnError(mFabricIndex.Value() == fabricIndex && mNodeId.Value() == nodeId, CHIP_ERROR_BUSY);
-
-            // Reset stale connection from the same Node if exists.
-            ResetState(CHIP_ERROR_CANCELLED);
-        }
-
-        auto * controller = [[MTRDeviceControllerFactory sharedInstance] runningControllerForFabricIndex:fabricIndex];
-        VerifyOrReturnError(controller != nil, CHIP_ERROR_INCORRECT_STATE);
-
-        mDelegate = controller.otaProviderDelegate;
-        mDelegateNotificationQueue = controller.otaProviderDelegateQueue;
-
-        // We should have already checked that this controller supports OTA.
-        VerifyOrReturnError(mDelegate != nil, CHIP_ERROR_INCORRECT_STATE);
-        VerifyOrReturnError(mDelegateNotificationQueue != nil, CHIP_ERROR_INCORRECT_STATE);
-
-        // Start a timer to track whether we receive a BDX init after a successful query image in a reasonable amount of time
-        CHIP_ERROR err = mSystemLayer->StartTimer(kBdxInitReceivedTimeout, HandleBdxInitReceivedTimeoutExpired, this);
-        LogErrorOnFailure(err);
-
-        ReturnErrorOnFailure(err);
-
-        mFabricIndex.SetValue(fabricIndex);
-        mNodeId.SetValue(nodeId);
-
-        mInitialized = true;
-
-        return CHIP_NO_ERROR;
-    }
-
-    bool mInitialized = false;
-    bool mTransferStarted = false;
-    Optional<FabricIndex> mFabricIndex;
-    Optional<NodeId> mNodeId;
-    id<MTROTAProviderDelegate> mDelegate = nil;
-    dispatch_queue_t mDelegateNotificationQueue = nil;
-    Messaging::ExchangeManager * mExchangeMgr = nullptr;
-
-    // Since we are a singleton, we get reused across transfers, but also have
-    // async calls that can happen.  The transfer generation keeps track of
-    // which transfer we are currently doing, so we can ignore async calls
-    // attached to no-longer-running transfers.
-    uint64_t mTransferGeneration = 0;
-};
-
-namespace {
-Global<BdxOTASender> gOtaSender;
-
-NSInteger constexpr kOtaProviderEndpoint = 0;
+NSInteger const kOtaProviderEndpoint = 0;
 } // anonymous namespace
 
-MTROTAProviderDelegateBridge::MTROTAProviderDelegateBridge() { Clusters::OTAProvider::SetDelegate(kOtaProviderEndpoint, this); }
+MTROTAProviderDelegateBridge::MTROTAProviderDelegateBridge()
+{
+    Clusters::OTAProvider::SetDelegate(kOtaProviderEndpoint, this);
+}
 
 MTROTAProviderDelegateBridge::~MTROTAProviderDelegateBridge()
 {
-    gOtaSender->ResetState(CHIP_ERROR_CANCELLED);
     Clusters::OTAProvider::SetDelegate(kOtaProviderEndpoint, nullptr);
 }
 
 CHIP_ERROR MTROTAProviderDelegateBridge::Init(System::Layer * systemLayer, Messaging::ExchangeManager * exchangeManager)
```

Comment:
i intended to pass it to mOtaUnsolicitedBDXMsgHandler actually. let me do that. And i did have code to do that but i removed it since i was not using systemlayer. but it makes sense to pass it.

Reply:
done.